pyartcd/pyartcd/pipelines/promote.py

```diff
@@ -375,6 +401,215 @@ def _reraise_if_not_permitted(self, err: VerificationError, code: str, permits:
         self._logger.warn("Issue %s is permitted with justification: %s", err, justification)
         return justification
 
+    async def publish_client(self, working_dir, from_release_tag, release_name, arch, client_type):
+        _, minor = util.isolate_major_minor_in_group(self.group)
+        quay_url = constants.QUAY_RELEASE_REPO_URL
+        # Anything under this directory will be sync'd to the mirror
+        BASE_TO_MIRROR_DIR = f"{working_dir}/to_mirror/openshift-v4"
+        shutil.rmtree(f"{BASE_TO_MIRROR_DIR}/{arch}", ignore_errors=True)
+
+        # From the newly built release, extract the client tools into the workspace following the directory structure
+        # we expect to publish to mirror
+        CLIENT_MIRROR_DIR = f"{BASE_TO_MIRROR_DIR}/{arch}/clients/{client_type}/{release_name}"
+        os.makedirs(CLIENT_MIRROR_DIR)
+
+        # extract release clients tools
+        extract_release_client_tools(f"{quay_url}:{from_release_tag}", f"--to={CLIENT_MIRROR_DIR}", None)
+
+        # Get cli installer operator-registory pull-spec from the release
+        for tarball in ["cli", "installer", "operator-registry"]:
+            image_stat, cli_pull_spec = get_release_image_pullspec(f"{quay_url}:{from_release_tag}", tarball)
+            if image_stat == 0:  # image exists
```

Comment:
Actually this check is not strictly correct. If `image_stat == 0`, we know the image must exist. However, if not, it doesn't mean the image doesn't exist. Other errors (e.g. network failure or auth error) could cause non-zero image_stat too.

Since this matches the existing logic, @Ximinhan could you add a `FIXME` comment here so we won't forget to fix this?

Reply:
added else case

Reply:
if it's other errors which is similar to oc shell command return status not 0 then it will raise in oc lib and will not get here